### ISSUE TITLE
how to hide country name code ?

### ISSUE THREAD

**Issue by zaheerpoei** [hbb20/CountryCodePickerProject#60]
i want to show only phone code. there is a method  showflag(boolean) that hides the flag if pass false to this. Like wise i want to hide country name code so there left only phone code. so there should be option to hide country code. Please tell me how can i hide country code. Thanks

**Comment by hbb20** [comment]
Definitely you can do it. Please check
https://github.com/hbb20/CountryCodePickerProject/blob/master/README.md#9-hide-country-name-code
.

**Comment by zaheerpoei** [comment]
thanks brother

**Comment by zaheerpoei** [comment]
good work keep it up

**Comment by zaheerpoei** [comment]
can we align the text as centre after selecting value of spinner ??

**Comment by hbb20** [comment]
You want to support full width right?

Right now I'm working on it. I will notify once it's implemented.

**Comment by zaheerpoei** [comment]
after selecting the value from dropdown it shows the selected value in spinner which is left aligned i want to align this to centre. thats it

**Comment by hbb20** [comment]
Can you send current screen and sketch of expected result?

**Comment by zaheerpoei** [comment]
![screenshot_2017-06-02-12-34-19 1](https://cloud.githubusercontent.com/assets/16101169/26715846/a9837bb4-4790-11e7-9f0f-e1daa9f89b84.png)

**Comment by hbb20** [comment]
If you can send me xml, atleast of between name edit text and button part,
I will try to set it.

If something required to be changed in library, I can take care in next
version.

**Comment by zaheerpoei** [comment]
<EditText
        android:id="@+id/name"
        android:layout_width="wrap_content"
        android:layout_height="44dp"
        android:layout_alignEnd="@+id/section_label"
        android:layout_alignLeft="@+id/section_label"
        android:layout_alignRight="@+id/section_label"
        android:layout_alignStart="@+id/section_label"
        android:layout_below="@+id/section_label"
        android:layout_marginTop="21dp"
        android:background="@drawable/shape"
        android:ems="10"
        android:padding="10dp"
        android:hint="Enter your first name"
        android:inputType="textPersonName" />


    <LinearLayout
        android:id="@+id/l1"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_alignBottom="@id/editText"
        android:layout_below="@+id/name"
        android:layout_marginTop="24dp"
        android:orientation="horizontal"
        android:background="@drawable/shape"
        android:weightSum="1">

        <com.hbb20.CountryCodePicker
            android:id="@+id/ccp"
            android:layout_width="match_parent"
            android:layout_height="44dp"
            android:layout_weight="0.8"


            app:hideNameCode="true" />


        <EditText
            android:id="@+id/number"
            android:layout_width="match_parent"
            android:layout_height="44dp"
            android:background="@drawable/shape"
            android:layout_weight="0.2"
            android:hint="Enter Your Cell Number"
            android:inputType="phone"
            android:padding="10dp" />
    </LinearLayout>


    <Button
        android:id="@+id/button"
        android:layout_width="wrap_content"
        android:layout_height="55dp"
        android:layout_marginTop="21dp"
        android:textAllCaps="false"
        android:backgroundTint="?attr/colorPrimary"
        android:text="Next"
        android:textColor="@android:color/white"
        android:layout_below="@+id/l1"
        android:layout_alignParentLeft="true"
        android:layout_alignParentStart="true"
        android:layout_alignParentRight="true"
        android:layout_alignParentEnd="true" />

**Comment by hbb20** [comment]
If you want to use weight sum then you should use width = 0. But still this approach is not the best.

Just replace your **l1** linear layout with 

```
<RelativeLayout
   android:id="@+id/rl"
   android:layout_width="match_parent"
   android:layout_height="44dp"
   android:layout_below="@+id/name"
   android:layout_marginTop="24dp"
   android:background="@drawable/shape"
   android:layout_alignBottom="@id/editText">

        <RelativeLayout
            android:id="@+id/rl_ccp_holder"
            android:layout_width="wrap_content"
            android:layout_height="match_parent"
            android:gravity="center_vertical">

            <com.hbb20.CountryCodePicker
                android:id="@+id/ccp"
                android:layout_width="wrap_content"
                android:layout_height="wrap_content"
                android:layout_centerVertical="true"
                app:hideNameCode="true"
                app:showFlag="false"/>
        </RelativeLayout>

        <EditText
            android:id="@+id/number"
            android:layout_width="match_parent"
            android:layout_height="match_parent"
            android:layout_toRightOf="@+id/rl_ccp_holder"
            android:background="@drawable/shape"
            android:hint="Enter Your Cell Number"
            android:inputType="phone"
            android:padding="10dp"/>
</RelativeLayout>
```

You will get correct result as per screen shot:
![layout-2017-06-02-205318](https://cloud.githubusercontent.com/assets/4918760/26732829/99829150-47d6-11e7-9d2e-a5dfe26d59c2.png)

**Comment by MrBrown09** [comment]
Hi,

Is it possible to restrict some country not to show in list?

**Comment by hbb20** [comment]
Yeah. You got it correct. But I would like to add something here.

Property `app:ccp_customMasterCountries` is used when you need to **_include_** some specific countries. [Read More](https://github.com/hbb20/CountryCodePickerProject/wiki/Custom-Master-Country-List)

If you want to **_exclude_** just a couple of countries, you should use
`app:ccp_excludedCountries`. [Read more](https://github.com/hbb20/CountryCodePickerProject/wiki/Exclude-specific-countries)

**Comment by josuesantamaria** [comment]
Hi, great library, thanks

Is it possible to hide the flags when you open the modal?

These flags: https://screencast.com/t/DmUzVJWz9

**Comment by hbb20** [comment]
@josuesantamaria , this was recently asked on https://github.com/hbb20/CountryCodePickerProject/issues/189.

Currently, there is no option for that. hopefully, we will be releasing new version this weekend with this option. Will that work?

**Comment by josuesantamaria** [comment]
It sounds great. Thank you very much. 

Also, would be great if we can have more control over the modal UI, like the width of the modal, put lines between the different options, make modal corners rounded. That kind of stuff, but thanks for your very helpful library.

**Comment by hbb20** [comment]
Try v2.2.1 and use `ccpDialog_showFlag` property to hide the flag from dialog.

Regarding your last comment. These requirements are very specific. We might not be able to incorporate it. If you want customized layout simply create a copy of [dialog layout](https://github.com/hbb20/CountryCodePickerProject/blob/master/ccp/src/main/res/layout/layout_picker_dialog.xml) with the same name in your layouts directory. You can change style in this copy. Note: do not change/remove id of views. At the run time android will pick your layout file instead of CCP's layout file. Hope this will be helpful.

**Comment by josuesantamaria** [comment]
It works perfectly! thank you very much and thanks for the information. Very good job!

**Comment by josuesantamaria** [comment]
Hi @hbb20 

I followed your advice about creating a copy of [dialog layout](https://github.com/hbb20/CountryCodePickerProject/blob/master/ccp/src/main/res/layout/layout_picker_dialog.xml), I have this:

```<?xml version="1.0" encoding="utf-8"?>
<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
    android:id="@+id/rl_holder"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:orientation="vertical"
    android:background="@drawable/country_code_picker_bg"
    android:padding="8dp">

    <RelativeLayout
        android:id="@+id/rl_title"
        android:layout_width="match_parent"
        android:layout_height="wrap_content">

        <TextView
            android:id="@+id/textView_title"
            android:layout_width="0dp"
            android:layout_height="0dp" />

        <ImageView
            android:id="@+id/img_dismiss"
            android:layout_width="36dp"
            android:layout_height="36dp"
            android:layout_alignParentStart="true"
            android:layout_alignParentLeft="true"
            android:layout_centerVertical="true"
            android:alpha="0.7"
            android:background="@drawable/ccp_selectable_bg"
            android:clickable="true"
            android:padding="4dp"
            android:src="@drawable/ic_clear_black_24dp" />

        <RelativeLayout
            android:id="@+id/rl_query_holder"
            android:layout_width="match_parent"
            android:layout_height="wrap_content"
            android:layout_margin="10dp"
            android:layout_toRightOf="@+id/img_dismiss"
            android:layout_toEndOf="@+id/img_dismiss">

            <EditText
                android:id="@+id/editText_search"
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
                android:layout_toLeftOf="@+id/img_clear_query"
                android:layout_toStartOf="@+id/img_clear_query"
                android:hint="@string/country_code_picker_search_hint"
                android:gravity="center"
                android:imeOptions="actionSearch"
                android:singleLine="true"
                android:inputType="textNoSuggestions|textVisiblePassword"
                android:textColor="@android:color/primary_text_light_nodisable" />

            <ImageView
                android:id="@+id/img_clear_query"
                android:layout_width="36dp"
                android:layout_height="36dp"
                android:layout_alignParentEnd="true"
                android:layout_alignParentRight="true"
                android:layout_centerVertical="true"
                android:alpha="0.7"
                android:background="@drawable/ccp_selectable_bg"
                android:clickable="true"
                android:padding="10dp"
                android:src="@drawable/ic_backspace_black_24dp" />
        </RelativeLayout>
    </RelativeLayout>

    <android.support.v7.widget.RecyclerView
        android:id="@+id/recycler_countryDialog"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:layout_below="@+id/rl_title"
        android:dividerHeight="2dp" />

    <com.futuremind.recyclerviewfastscroll.FastScroller
        android:id="@+id/fastscroll"
        android:layout_width="wrap_content"
        android:layout_height="match_parent"
        android:layout_alignBottom="@+id/recycler_countryDialog"
        android:layout_alignEnd="@+id/recycler_countryDialog"
        android:layout_alignRight="@+id/recycler_countryDialog"
        android:layout_alignTop="@+id/recycler_countryDialog"
        android:orientation="vertical" />

    <TextView
        android:id="@+id/textView_noresult"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_below="@+id/rl_title"
        android:layout_centerHorizontal="true"
        android:layout_marginTop="16dp"
        android:text="@string/no_result_found"
        android:textColor="@android:color/primary_text_light"
        android:visibility="gone" />

</RelativeLayout>
```

Like you can see, I just accommodated some fields and changed the `editText_search` hint value, I also tried to remove the `textView_title` but it gives me an error and I had to keep it as an empty value.

The problem that I'm facing is that even that I changed the hint value, the text is not updated, it seems that the text is overwritten by some piece of code inside the library, Do you know what is happening? Also, I tried to make a copy of [country tile](https://github.com/hbb20/CountryCodePickerProject/blob/master/ccp/src/main/res/layout/layout_recycler_country_tile.xml) to change the text colors, but again, it seems to be overwritten by another piece of code.

I would appreciate a lot your help

**Comment by hbb20** [comment]
- Hint text is coming from the library because for different languages this will be different.
- To hide the title, you might wish to use `app:ccpDialog_showTitle="false"` [Read more](https://github.com/hbb20/CountryCodePickerProject/wiki/XML-Properties#appccpdialog_showtitlefalse--default-true)

Regarding the country tile, I will check and post back.

Do you mind creating a new issue as this part is not alligned with original issue.

**Comment by shukla12manish** [comment]
Hi @hbb20 
There is a option for flag hiding using 

`app:showFlag="false"`

There is a option for hiding country name code using

 `app:hideNameCode="true"`

**But i want to show only flag and selection arrow and hide country code and country name. I am able to hide country name code but not able to hide country code. How can i do it ?**

**Comment by hbb20** [comment]
What is the version of library you are using?

**Comment by shukla12manish** [comment]
@hbb20  I followed the steps given in this link 

> https://jitpack.io/p/inpossoft/CountryCodePickerProject


version: 2.0.0

`implementation 'com.github.inpossoft:CountryCodePickerProject:2.0.0'`

**Comment by shukla12manish** [comment]
@hbb20 I think i have used different library. Can you tell me how to add your library into my project step by step, as i tried to add 
```
dependencies {
          implementation 'com.hbb20:ccp:2.2.4'
        }
```

but it gave me error.

**Comment by hbb20** [comment]
For integration, https://github.com/hbb20/CountryCodePickerProject/wiki/How-to-integrate-into-your-project should be helpful. You get some problem integrating that, feel free to comment back. We will be happy to help you with it.

**Comment by hbb20** [comment]
Once your setup is complete, use following properties to show only flag.
![image](https://user-images.githubusercontent.com/4918760/53702835-e8d0ca00-3dd0-11e9-93cf-dc047062d15a.png)

**Comment by shukla12manish** [comment]
@hbb20 Hi thanks for the reply but i am still stuck on 3 and 4th step in implementing the library.

Add CCP object in Activity / Fragment
  `CountryCodePicker ccp;`

Bind CCP from layout
 ` ccp = (CountryCodePicker) findViewById(R.id.ccp);`
 
where should i place these codes ?

after adding dependency i synced but android studio showed me 

> Android resource linking failed

**Comment by hbb20** [comment]
Did you remove 
    implementation 'com.github.inpossoft:CountryCodePickerProject:2.0.0' import?

**Comment by m7md7sn994** [comment]
Hi 
How to add padding ?

**Comment by hbb20** [comment]
@m7md7sn994 sir, can you please create a new issue with little more explanation about where you want to add padding?
